internal/orchestrator/updates_test.go

```diff
@@ -146,3 +149,62 @@ func TestInstallEventFromKV(t *testing.T) {
 	_, err = installEventFromKV(context.Background(), entry)
 	require.ErrorIs(t, errStaleEvent, err)
 }
+
+func TestConditionListenersExit(t *testing.T) {
+	srv := startJetStreamServer(t)
+	defer shutdownJetStream(t, srv)
+	nc, _ := jetStreamContext(t, srv) // nc is closed on evJS.Close(), js needs no cleanup
+	evJS := events.NewJetstreamFromConn(nc)
+	defer evJS.Close()
+
+	log := logrus.New()
+	defs := ptypes.ConditionDefinitions{
+		&ptypes.ConditionDefinition{
+			Kind: ptypes.FirmwareInstall,
+		},
+		&ptypes.ConditionDefinition{
+			Kind: ptypes.InventoryOutofband,
+		},
+		&ptypes.ConditionDefinition{
+			Kind: ptypes.ConditionKind("bogus"),
+		},
+	}
+	// XXX: THIS FUNCTION CAN ONLY BE CALLED ONCE IN THE ENTIRE TEST! It uses a sync.Once
+	status.ConnectToKVStores(evJS, log, defs, kv.WithDescription("test watchers"))
+
+	o := Orchestrator{
+		logger:        log,
+		streamBroker:  evJS,
+		facility:      "test",
+		conditionDefs: defs,
+	}
+
+	// here we're acting in place of kvStatusPublisher to make sure that we can orchestrate the watchers
+	var wg sync.WaitGroup
+	testChan := make(chan *v1types.ConditionUpdateEvent)
+	ctx, cancel := context.WithCancel(context.TODO())
+
+	o.startConditionListeners(ctx, testChan, &wg)
+
+	sentinelChan := make(chan struct{})
+	toCtx, toCancel := context.WithTimeout(context.TODO(), time.Second)
+	defer toCancel()
+
+	var testPassed bool
+
```

Comment:
Would be worth including a goroutine leak check in this test https://github.com/uber-go/goleak#quick-start

Reply:
TIL. I'll look into it, because that's definitely something I want to avoid here.